### PR TITLE
chore: use Rust `1.80.0`

### PR DESCRIPTION
**Pull request by EstebanBorai**
Updates Rust Toolchain to `1.80.0` and applies clippy improvements

### CODE REVIEW

**Review by morenol — APPROVED** [review]

**Review by sehz — APPROVED** [review]

**Comment by EstebanBorai** [comment]
@morenol, @sehz I will need a second review on this PR because I had to remove `stats` feature which files are no longer in `fluvio` crate.

I did these changes on: [4fd5986](https://github.com/infinyon/fluvio/pull/4118/commits/4fd598603fe939ca2b308500f0f667d2e84cda1f).

**Review by sehz — APPROVED** [review]

**Pull request merged**
